### PR TITLE
chore: smoke test web apps dependencies before running functional tests

### PR DESCRIPTION
**Pull request by SebFlippence**
### Change description ###

Check that services the web app frontend relies on are up before running the functional tests.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/smoke/smoke.ts [hunk 1]
```diff
@@ -2,26 +2,34 @@ import axios from 'axios';
...
-jest.retryTimes(20); // 20 retries at 3 second intervals = 1 minute
+jest.retryTimes(20); // 20 retries at 1 second intervals
 jest.setTimeout(15000);
 
+const servicesToCheck = [
+  { name: 'No Fault Divorce Web', url: config.TEST_URL },
+  { name: 'IDAM Web', url: 'https://idam-web-public.aat.platform.hmcts.net' },
+  { name: 'IDAM API', url: 'https://idam-api.aat.platform.hmcts.net' },
+  { name: 'Auth Provider', url: 'http://rpe-service-auth-provider-aat.service.core-compute-aat.internal' },
+  { name: 'CCD Data Store', url: 'http://ccd-data-store-api-aat.service.core-compute-aat.internal' },
```
[the remark this answers is not in the document]
Comment (SebFlippence) [reply src/test/smoke/smoke.ts:13]: These seem to be the all of services that the app is communicating with or redirecting to on the preview environments. If any of these are down during the smoke test, then there's not really much point continuing to run the functional tests as they will fail.

Which one's do you think we shouldn't be checking?

[the remark this answers is not in the document]
Comment (SebFlippence) [reply src/test/smoke/smoke.ts:13]: These are the 4 services (+ the app itself) that are listed in the [config/default.yaml](https://github.com/hmcts/nfdiv-frontend/blob/master/config/default.yaml). If any of these are down then there's not too much point running the functional tests as they will most likely fail.

Which services do you think we shouldn't be checking?

[the remark this answers is not in the document]
Comment (linusnorton) [reply src/test/smoke/smoke.ts:13]: Well technically they should be included in our health check but we got discouraged from doing that as it hammered those services. So I guess what you've got here is fine

[the remark this answers is not in the document]
Comment (SebFlippence) [reply src/test/smoke/smoke.ts]: 👍 Good point, I've updated this to pull in all of the URLs from the `yaml` config in 8e321a3.

**Pull request merged**